### PR TITLE
ols: init at nightly-2023-05-18

### PR DESCRIPTION
**Pull request by Astavie**
###### Description of changes

https://github.com/DanielGavin/ols

commit `fd13619` has been cherry-picked because it is the last version that compiles with the latest release version of Odin. Later commits require Odin to be compiled from the master branch. ols also doesn't have any versioned releases yet.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/ols/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+stdenv.mkDerivation {
+  pname = "ols";
+  version = "20230421";
+
+  src = fetchFromGitHub {
+    owner = "DanielGavin";
+    repo = "ols";
+    rev = "fd136199897d5e5c87f6f1fbfd076ed18e41d7b7";
+    sha256 = "sha256-lRoDSc2bZSuXTam3Q5OOlSD6YAobCFKNRbtQ41Qx5EY=";
```
Comment (drupol) [inline pkgs/development/tools/ols/default.nix:11]:
```suggestion
    hash = "sha256-lRoDSc2bZSuXTam3Q5OOlSD6YAobCFKNRbtQ41Qx5EY=";
```

File: pkgs/development/tools/ols/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, makeWrapper, odin, lib }:
```
Comment (drupol) [inline pkgs/development/tools/ols/default.nix:1]: Prefer `makeBinWrapper` to `makeWrapper`. It prevent issues on Darwin platforms.

File: pkgs/development/tools/ols/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  buildPhase = ''
+    ./build.sh
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ols $out/bin
+    wrapProgram $out/bin/ols --set-default ODIN_ROOT ${odin}/share
+  '';
```
Comment (drupol) [inline pkgs/development/tools/ols/default.nix:34]:
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out/bin
    cp ols $out/bin
    wrapProgram $out/bin/ols --set-default ODIN_ROOT ${odin}/share

    runHook postInstall
  '';
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/ols/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, makeWrapper, odin, lib }:
+
+stdenv.mkDerivation {
+  pname = "ols";
+  version = "20230421";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ols/default.nix:5]:
```suggestion
  version = "unstable-2023-04-21";
```

File: pkgs/development/tools/ols/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  buildInputs = [
+    odin
+  ];
+
+  postPatch = ''
+    patchShebangs build.sh
+  '';
+
+  buildPhase = ''
+    ./build.sh
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ols/default.nix:27]:
```suggestion
    odin build src/ -collection:shared=src -out:ols -o:speed
```
this sounds like it is going to do native optimisation. We must turn this off.

Comment (Astavie) [reply pkgs/development/tools/ols/default.nix:27]: it won't - you'd need to pass the `-microarch:native` flag to odin for that to happen

File: pkgs/development/tools/ols/default.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchFromGitHub, makeBinaryWrapper, odin, lib }:
+
+stdenv.mkDerivation {
+  pname = "ols";
+  version = "nightly-2023-05-18";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ols/default.nix:5]:
```suggestion
  version = "unstable-2023-05-18";
```

**Pull request merged**